api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java

```diff
@@ -151,7 +152,19 @@ public DbUser linkRasLoginGovAccount(String authCode, String redirectUrl) {
 
     // If eRA is not already linked, check response from RAS see if RAS contains eRA Linking
     // information.
-    DbUser user = userService.updateRasLinkLoginGovStatus(getLoginGovUsername(userInfoResponse));
+    String username = getUsername(userInfoResponse);
```

Comment:
`getLoginGovUsername` used to see if userInfo contained a login.gov for its preferredUsername and threw an an exception otherwise.

I broke up this logic so that `getUsername` now returns the preferredUsername field. Then the if block within `linkRasAccount` checks whether the username is associated with id.me or login.gov. In either case, it updates the corresponding data access module. If it matches neither, an exception is thrown.